File: src/main/java/me/alegian/thavma/impl/init/registries/T7Registries.java
package me.alegian.thavma.impl.init.registries;

import me.alegian.thavma.impl.Thavma;
import me.alegian.thavma.impl.common.aspect.Aspect;
import me.alegian.thavma.impl.common.research.Research;
import me.alegian.thavma.impl.common.wand.WandCoreMaterial;
import me.alegian.thavma.impl.common.wand.WandHandleMaterial;
import net.minecraft.core.Registry;
import net.minecraft.resources.ResourceKey;
import net.neoforged.neoforge.registries.RegistryBuilder;

public class T7Registries {
  public static final Registry<WandHandleMaterial> WAND_HANDLE = new RegistryBuilder<WandHandleMaterial>(ResourceKey.createRegistryKey(Thavma.rl("wand_handle")))
      .maxId(Integer.MAX_VALUE)
      .create();

  public static final Registry<WandCoreMaterial> WAND_CORE = new RegistryBuilder<WandCoreMaterial>(ResourceKey.createRegistryKey(Thavma.rl("wand_core")))
      .maxId(Integer.MAX_VALUE)
      .create();

  public static final Registry<Aspect> ASPECT = new RegistryBuilder<Aspect>(ResourceKey.createRegistryKey(Thavma.rl("aspect")))
      .maxId(Integer.MAX_VALUE)
      .create();

  public static final Registry<Research> RESEARCH = new RegistryBuilder<Research>(ResourceKey.createRegistryKey(Thavma.rl("research")))
      .maxId(Integer.MAX_VALUE)
      .create();
}

File: src/main/java/me/alegian/thavma/impl/init/registries/T7Registries.kt
package me.alegian.thavma.impl.init.registries

import me.alegian.thavma.impl.Thavma.rl
import me.alegian.thavma.impl.common.aspect.Aspect
import me.alegian.thavma.impl.common.research.Research
import me.alegian.thavma.impl.common.wand.WandCoreMaterial
import me.alegian.thavma.impl.common.wand.WandHandleMaterial
import net.minecraft.resources.ResourceKey
import net.neoforged.neoforge.registries.RegistryBuilder

object T7Registries {
    val WAND_HANDLE = RegistryBuilder(ResourceKey.createRegistryKey<WandHandleMaterial>(rl("wand_handle")))
        .maxId(Int.MAX_VALUE)
        .create()

    val WAND_CORE = RegistryBuilder(ResourceKey.createRegistryKey<WandCoreMaterial>(rl("wand_core")))
        .maxId(Int.MAX_VALUE)
        .create()

    val ASPECT = RegistryBuilder(ResourceKey.createRegistryKey<Aspect>(rl("aspect")))
        .maxId(Int.MAX_VALUE)
        .create()

    val RESEARCH = RegistryBuilder(ResourceKey.createRegistryKey<Research>(rl("research")))
        .maxId(Int.MAX_VALUE)
        .create()
}
